source/audio/appsVolume.py

```diff
@@ -0,0 +1,160 @@
+		try:
+			with _appVolumesCacheLock:
+				originalVolumeAndMute = _appVolumesCache[pid]
+				del _appVolumesCache[pid]
+		except KeyError:
+			originalVolumeAndMute = VolumeAndMute(volume=1.0, mute=False)
+		return originalVolumeAndMute
+
+	def onSessionUpdate(self, session: AudioSession) -> None:
+		pid = session.ProcessId
+		simpleVolume = session.SimpleAudioVolume
+		with _appVolumesCacheLock:
+			if pid not in _appVolumesCache:
+				_appVolumesCache[pid] = VolumeAndMute(
+					volume=simpleVolume.GetMasterVolume(),
+					mute=simpleVolume.GetMute(),
+				)
+		if pid != globalVars.appPid:
+			simpleVolume.SetMasterVolume(self.volumeAndMute.volume, None)
+			simpleVolume.SetMute(self.volumeAndMute.mute, None)
+
+	def onSessionTerminated(self, session: AudioSession) -> None:
+		pid = session.ProcessId
+		simpleVolume = session.SimpleAudioVolume
+		originalVolumeAndMute = self.getOriginalVolumeAndMute(pid)
+		try:
+			simpleVolume.SetMasterVolume(originalVolumeAndMute.volume, None)
+			simpleVolume.SetMute(originalVolumeAndMute.mute, None)
+		except Exception:
+			log.exception(f"Could not restore master volume of process {pid} upon exit.")
+
+
+def _updateAppsVolumeImpl(
+		volume: float,
+		state: AppsVolumeAdjusterFlag,
+):
+	global _activeCallback
+	if state == AppsVolumeAdjusterFlag.DISABLED:
+		newCallback = DummyAudioSessionCallback()
+		runTerminators = True
+	else:
+		newCallback = VolumeSetter(
+			volumeAndMute=VolumeAndMute(
+				volume=volume,
+				mute=state == AppsVolumeAdjusterFlag.MUTED,
+			)
+		)
+		runTerminators = False
+	if _activeCallback is not None:
+		_activeCallback.unregister(runTerminators=runTerminators)
+	_activeCallback = newCallback
+	_activeCallback.register()
+
+
+def _adjustAppsVolume(
+		volumeAdjustment: int | None = None,
+):
+	if not nvwave.usingWasapiWavePlayer():
+		message = _(
+			# Translators: error message when wasapi is turned off.
+			"Other applications' volume cannot be adjusted. "
+			"Please enable WASAPI in the Advanced category in NVDA Settings to use it."
+		)
+		ui.message(message)
+		return
+	volume: int = config.conf["audio"]["applicationsSoundVolume"]
+	state = config.conf["audio"]["applicationsVolumeMode"]
+	if state != AppsVolumeAdjusterFlag.ENABLED:
+		# Translators: error message when applications' volume is disabled
+		msg = _("Please enable applications' volume adjuster in order to adjust applications' volume")
+		ui.message(msg)
+		return
+	volume += volumeAdjustment
+	volume = max(0, min(100, volume))
+	log.debug(f"Adjusting applications volume by {volumeAdjustment}% to {volume}%")
+	config.conf["audio"]["applicationsSoundVolume"] = volume
+
+	# We skip running terminators here to avoid application volume spiking to 100% for a split second.
+	_updateAppsVolumeImpl(volume / 100.0, state)
+	# Translators: Announcing new applications' volume message
+	msg = _("Applications volume {}").format(volume)
+	ui.message(msg)

```

Comment:
I disagree with the rabbit here.
1. We shouldn't check for values not being None just in case. This would create too much boilerplate code that is never executed and never tested and make code harder to read. Currently all calls to `_adjustAppsVolume()` have non-null argument passed.
2. Edge case check is currently done correctly. The rabbit proposes to just log error when boundaries check fails which is wrong.